Code/GraphMol/FileParsers/MolFileParser.cpp

```diff
@@ -1328,11 +1328,11 @@ void convertComplexNameToQuery(Atom *query, const std::string &symb) {
 }
 
 Atom *ParseMolFileAtomLine(const std::string text, RDGeom::Point3D &pos,
-                           unsigned int line) {
+                           unsigned int line, bool strictParsing) {
   std::string symb;
   int massDiff, chg, hCount;
 
-  if (text.size() < 34) {
+  if ((strictParsing && text.size() < 34) || text.size() < 32) {
```

Comment:
A few lines after this (github won't allow me to mark it) we have a line that says: 
```
symb = text.substr(31, 3);
```
This assumes we have 34 chars; we should adjust it to something like:
```
symb_sz = std::min(3, text.size() - 31);
symb = text.substr(31, symb_sz);
```

Reply:
I think we don’t have to worry about this since `substr()` will just stop if you ask for more characters than are available in the string:
http://www.cplusplus.com/reference/string/string/substr/

Reply:
Oh, ok, I never remember it is only the `pos` argument that causes an exception if it excedes the string length.